app/src/main/java/com/training/app/trainingapp/main/MainActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.training.app.trainingapp.main
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import com.training.app.trainingapp.R
+import com.training.app.trainingapp.main.view.RegisterPageView
+import com.training.app.trainingapp.main.view_model.RegisterViewModel
+
+
+class MainActivity : ComponentActivity() {
+
+    private val registerViewModel: RegisterViewModel by viewModels()
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+
+
+
+        setContent {
+            Surface(modifier = Modifier.fillMaxSize(), color = Color.White) {
+                val email = registerViewModel.userEmail.collectAsState().value
+                val emailStatus = registerViewModel.emailValidateState.collectAsState().value
+                RegisterPageView(email,
+                    emailStatus,
+                    onEmailChanged = {
+                        registerViewModel.onEmailChanged(it)
+                    },
+                    stringResource(id = R.string.check),
```

Comment:
here you can send button's text from a lower level, inside the RegisterPageView to the button

Reply:
it's correct